cmake/kokkos_functions.cmake

```diff
@@ -1,41 +1,30 @@
 ################################### FUNCTIONS ##################################
 # List of functions
-#   set_kokkos_cxx_compiler
-#   set_kokkos_cxx_standard
 #   kokkos_option
 
+# Validate options are given with correct case and define an internal
+# upper-case version for use within 
 FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
```

Comment:
Did you mean to ignore the `DOCSTRING`?  The help text is lost and the user is not able to see it in the cache.

Reply:
I moved all of the internal `KOKKOS_X` from being persistent cache variables to regular variables. This massively simplifies the logic of things when people re-configure, which is why this got lost.

But, you are right.  There should be a 
````
SET(${CAMEL_NAME} ${DEFAULT} ${TYPE} CACHE ${DOCSTRING})
````
The `Kokkos_X` input variables should show up in cache with doc strings.